e2e/tests/component-preview.spec.ts

```diff
@@ -6,122 +6,139 @@ const test = base.extend({
+        await expect(page.getByRole('tabpanel')).toContainText('<a href')
+        await expect(page.getByRole(tabRole, { name: 'HTML' })).toHaveAttribute('aria-expanded', 'true')
+        await expect(page.getByRole(tabRole, { name: 'Nunjucks' })).toHaveAttribute('aria-expanded', 'false')
+      })
 
-    test.describe('@visual-regression', () => {
-      test('matches the saved screenshot when the preview is resized', async ({ page }) => {
-        const boundingBox = await page.locator('.guidance-component-preview__preview').boundingBox()
+      test('does not change the URL', async ({ page }) => {
+        await expect(page).toHaveURL(/\/components\/back-link\/$/)
+      })
 
-        if (!boundingBox) {
-          throw new Error('boundingBox was null')
-        }
+      test('switches to the Nunjucks tab contents when that tab is clicked', async ({ page, tabRole }) => {
+        await page.getByRole(tabRole, { name: 'Nunjucks' }).click()
 
-        const boxRight = boundingBox.x + boundingBox.width
-        const boxBottom = boundingBox.y + boundingBox.height
+        await expect(page.getByRole('tabpanel')).toContainText('{{ modukBackLink({')
+        await expect(page.getByRole(tabRole, { name: 'Nunjucks' })).toHaveAttribute('aria-expanded', 'true')
+        await expect(page.getByRole(tabRole, { name: 'HTML' })).toHaveAttribute('aria-expanded', 'false')
+      })
 
-        await page.mouse.move(boxRight - 1, boxBottom - 1)
-        await page.mouse.down()
-        await page.mouse.move(boxRight - 100, boxBottom - 1)
-        await page.mouse.up()
+      test.describe('@visual-regression', () => {
+        const screenshotName = url.replace(/\/components\/(.*)\//g, '$1')
+        test('matches the saved screenshot when the preview is resized', async ({ page }) => {
+          const boundingBox = await page.locator('.guidance-component-preview__preview').boundingBox()
 
-        const componentPreview = page.locator('.guidance-component-preview')
-        await expect(componentPreview).toHaveScreenshot(
-          'component-preview-resized.png',
-        )
-      })
+          if (!boundingBox) {
+            throw new Error('boundingBox was null')
+          }
+
+          const boxRight = boundingBox.x + boundingBox.width
+          const boxBottom = boundingBox.y + boundingBox.height
 
-      const TABS = ['HTML', 'Nunjucks']
+          await page.mouse.move(boxRight - 1, boxBottom - 1)
+          await page.mouse.down()
+          await page.mouse.move(boxRight - 200, boxBottom - 1)
+          await page.mouse.up()
 
-      TABS.forEach((tabName) => {
-        test(`matches the saved screenshot when the ${tabName} tab is not hovered`, async ({ page }) => {
           const componentPreview = page.locator('.guidance-component-preview')
           await expect(componentPreview).toHaveScreenshot(
-            `component-preview-${tabName.toLowerCase()}-tab-unhovered.png`,
+            [screenshotName, 'component-preview-resized.png'],
```

Comment:
`screenshotName` is using regex and separating them into folders 
`const screenshotName = url.replace(/\/components\/(.*)\//g, '$1')`